### PR TITLE
Minimal backend typescript support

### PR DESCRIPTION
**Pull request by partouf**
This is mostly copied from Austin's typescript PR

Note:
* tsc is run in the makefile, but it's really just for compilation checks, we don't need the generated .js files
* We'll need to change infra to do `-r ts-node/register` before this can go live

### CODE REVIEW

File: .mocharc.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+recursive: true
```
Comment (mattgodbolt) [inline .mocharc.yml:1]: does this duplicate the stuff in `package.json` (which also lists `require: `esm`, as well as `test/_setup` not listed here)

File: .nycrc.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 all: true
 include:
   - lib/**/*.js
+  - lib/**/*.ts
```
Comment (mattgodbolt) [inline .nycrc.yml:4]: :+1:

File: app.js [hunk 3]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
 // shebang interferes with license header plugin
```
Comment (mattgodbolt) [inline app.js:1]: having deleted the shebang we can remove the commant and disable I think?

Comment (partouf) [reply app.js:1]: ah right

File: .mocharc.yml (hunk 1 above)
Comment (partouf) [reply .mocharc.yml:1]: it was required for `npm run test`, because `node_modules/.bin/mocha` runs node independently

**Comment by partouf** [comment]
Beepboop! Typescript hello! Thanks @apmorton for the work which made this possible!

**Pull request merged**